librepo/yum.c

```diff
@@ -152,9 +161,23 @@ lr_yum_repomd_record_enabled(LrHandle *handle, const char *type)
                 return TRUE;
             x++;
         }
+        // Substitution check
+        if (handle->yumslist) {
+            for (GSList *elem = handle->yumslist; elem; elem = g_slist_next(elem)) {
+                LrVar* subs = elem->data;
+                if (!g_strcmp0(subs->val, type)) {
+                    char *orig = subs->var;
+                    for (guint i = 0; handle->yumdlist[i]; i++) {
```

Comment:
This whole thing is inside `if (handle->yumdlist)` so there is no need to check it again.